tests/cpp/test_circular_buffering.cpp

```diff
@@ -863,8 +863,11 @@ class TmaCircularBufferingTest
     auto reference_cpu = reference_cpu_data.accessor<data_type, 1>();
     auto result_cpu = result_cpu_data.accessor<data_type, 1>();
 
-    constexpr double tolerance = 1e-3;
+    constexpr double abs_tolerance = 1e-3;
+    constexpr double rel_tolerance = 1e-3;
```

Comment:
The outer reduction kernel in the new test uses serial reduction, which by its nature has larger error. So we need to consider relative tolerance as well to avoid false positive.